Adding Basic Peribolos functionality

aepfli opened the pull request:
Adding basic configuration (not populated) and a basic generator script, based on kubernetes/org to showcase our usage and to open it up for a better discussion.

Closes: #220

Signed-off-by: Simon Schrottner <simon.schrottner@dynatrace.com>

AloisReitbauer commented:
Would you please provide a bit more insight on this for the GC.

aepfli commented:
> Would you please provide a bit more insight on this for the GC.

the main information is located in #220 - where we outline the approach and the goals. If you do have questions/concerns etc. we should best discuss this in the issue - as this is currently only a draft to showcase the outline and basic implementation.

mowies commented on COMMUNITY_MEMBERSHIP.md:
```diff
@@ -81,7 +91,7 @@ Members are continuously active contributors in the community. They can have iss
 Code approvers are able to both review and approve code contributions, as well as help maintainers triage issues and do project management. While code review is focused on code quality and correctness, approval is focused on holistic acceptance of a contribution including: backwards/forwards compatibility, adhering to API and flag conventions, subtle performance and correctness issues, interactio ...
 
-*Defined by:*  [Approvers of the Keptn GitHub organization](https://github.com/orgs/keptn/teams/approvers), and [CODEOWNERS](https://help.github.com/en/articles/about-code-owners) in [Keptn](https://github.com/keptn/keptn)
+*Defined by:* Member of the subproject approvers team (https://github.com/orgs/keptn/teams/<subproject>-approvers) and listed in the subprojects `teams.yaml` within [Peribolos Configuration Directory][Peribolos Configuration Directory]
```
same thing here with the half-rendered link

aepfli replied:
thank you changed it to code

mowies commented on COMMUNITY_MEMBERSHIP.md:
```diff
@@ -118,14 +128,12 @@ Approver status can be scoped to a part of the codebase. For example, critical c
-* [Maintainers of the Keptn GitHub organization](https://github.com/orgs/keptn/teams/maintainers)
-* Ownership of the GitHub repository
-* Permissions and entry in `CODEOWNERS` files
+* Member of the subproject maintainer team (https://github.com/orgs/keptn/teams/<subproject>-maintainer) and listed in the subprojects `teams.yaml` within [Peribolos Configuration Directory][Peribolos Configuration Directory]
```
same link problem here

aepfli replied:
thank you changed it to code

mowies commented on COMMUNITY_MEMBERSHIP.md:
```diff
@@ -162,5 +171,56 @@ The process for becoming a maintainer should be defined in the project or subpro
+
+## Maintainer
+
+Maintainers are the technical authority for all subprojects within the Keptn project. They *MUST* have demonstrated both good judgment and responsibility towards the health of the Keptn project. Maintainers *MUST* set technical direction and make or approve design decisions for those subproject; either directly or through delegation of these responsibilities.
```
```suggestion
Maintainers are the technical authority for all subprojects within the Keptn project. They *MUST* have demonstrated both good judgment and responsibility towards the health of the Keptn project. Maintainers *MUST* set technical direction and make or approve design decisions for the overall project; either directly or through delegation of these responsibilities.
```

mowies commented on COMMUNITY_MEMBERSHIP.md:
```diff
@@ -162,5 +171,56 @@ The process for becoming a maintainer should be defined in the project or subpro
+
+### Requirements
+
+As `Maintainer` has the same requirements as a `subproject Maintainer` but those apply on a global-level.
```
```suggestion
As `Maintainer` has the same requirements as a `subproject Maintainer` but those apply on a global level.
```

aepfli commented on .github/workflows/peribolos.yaml:
```diff
@@ -0,0 +1,26 @@
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
good catch, this was for testing the workflow, to see if it is actually running - we do not need it on prs will change

aepfli commented on .github/ISSUE_TEMPLATE/membership.md:
```diff
@@ -64,4 +68,10 @@ Please remember, it is an applicant's responsibility to get their sponsors' conf
\ No newline at end of file
+- SIG projects I am involved with
+
+## References
+
+- [Community Membership Guidelines]: https://github.com/keptn/community/blob/master/COMMUNITY_MEMBERSHIP.md
+- [Peribolos Configuration]: ./config/keptn/org.yaml
+- [Peribolos Configuration Directory]: ./config/keptn/
```
```suggestion

[Community Membership Guidelines]: https://github.com/keptn/community/blob/master/COMMUNITY_MEMBERSHIP.md
[Peribolos Configuration]: ./config/keptn/org.yaml
[Peribolos Configuration Directory]: ./config/keptn/

```

aepfli commented on .github/ISSUE_TEMPLATE/membership.md:
```diff
@@ -64,4 +68,10 @@ Please remember, it is an applicant's responsibility to get their sponsors' conf
+
+- [Community Membership Guidelines]: https://github.com/keptn/community/blob/master/COMMUNITY_MEMBERSHIP.md
+- [Peribolos Configuration]: ./config/keptn/org.yaml
+- [Peribolos Configuration Directory]: ./config/keptn/
```
yes they should not have been a list, i changed it for the membership but not here. nice catch again

mowies commented on .github/workflows/peribolos.yaml:
```diff
@@ -15,17 +16,18 @@ jobs:
       - name: Run Peribolos
         uses: docker://ghcr.io/dynatrace-innovationlab/peribolos:v0
         with:
+          # TODO add confirm before merge
```
does this mean that we could have a pipeline that just tells us what it would change on PRs and then the confirm flag(?) on the merge trigger?

aepfli replied:
Yes, we could - I thought about that too - but we should also limit this workflow to changes within `config` and `tools` - what would be your best suggestion to define this? I still think https://stackoverflow.com/a/63131743/3708208 is somewhat of an ugly solution - but maybe you have a better one?

mowies replied:
you can define triggers to be limited to certain paths as well. see https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#example-including-paths

aepfli commented on .github/workflows/peribolos.yaml:
```diff
@@ -0,0 +1,42 @@
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
```suggestion
  pull_request:
    branches: [ main ]
    paths:
      - 'tools/**'
      - 'config/**'
```

mowies commented on config/keptn/utils/teams.yaml:
```diff
@@ -0,0 +1,14 @@
+repos:
+  - gh-action-extract-branch-name
+  - keptn-on-keptn
```
this repo can be deleted. it would have been under the ops team, but it's not needed anymore

aepfli commented:
@AnaMMedina21 @AloisReitbauer @AlexsJones 

This is a critical pr that needs the approval of the governance board - I am not sure if one committee approval is enough for this.

I changed the membership ladder and enhanced it with this current status of two different projects - hence, this is a change to the structure of this community, and we need the full support of the governance committee for this.

aepfli commented:
i renamed the `teams.yaml` files to `workgroup.yaml` to reduce confusion with github teams and our teams (so we call them workgroups)

Additionally i added some logic to provide defaults for certain repo settings - so that we do not need to define them all the time.

thisthat reviewed (approved):
Thanks @aepfli
